examples/IIM/ex2/example.cpp

```diff
@@ -0,0 +1,1237 @@
+                    {
+						p_ex_qp = 0.0;
+                    }    
+                    else if (sqrt((y - 0.5*H)*(y - 0.5*H) + (z - 0.5*H)*(z - 0.5*H)) < D/2 - patch_dx[1])
+                    { 
+						p_ex_qp = -2.0*p_e*x /L + p_e;
+						
+					}
+					else
+					{
+						p_ex_qp = p1;
+					}
+                   
+                    if (x > 0.2*L && x < 0.8*L) 
+					{
+						qp_tot +=1;
+						p_Eulerian_L2_norm += std::abs(p1 - p_ex_qp) * std::abs(p1 - p_ex_qp) * (*wgt_cc_data)(cell_idx);
+						p_Eulerian_max_norm = std::max(p_Eulerian_max_norm, std::abs(p1 - p_ex_qp));
+					}
+                }
+            }
+                    
+        }
+    }
+		SAMRAI_MPI::sumReduction(&qp_tot, 1);
+        SAMRAI_MPI::sumReduction(&p_Eulerian_L2_norm, 1);
+        SAMRAI_MPI::maxReduction(&p_Eulerian_max_norm, 1);
+ 
+        p_Eulerian_L2_norm = sqrt(p_Eulerian_L2_norm);
+        
+         pout << " p_Eulerian_L2_norm = " << p_Eulerian_L2_norm <<"\n\n";
+         pout << " p_Eulerian_max_norm = " << p_Eulerian_max_norm <<"\n\n";
+
+
+    return;
+} // compute_pressure_profile
+
+
+void compute_flow_rate(const double dt,
+                       const Pointer<PatchHierarchy<NDIM> > hierarchy,
+                       const int U_idx,
+                       const double loop_time,
+                       const int wgt_sc_idx)
+{
+	vector<double> qsrc;
+	qsrc.resize(2);
+	std::fill(qsrc.begin(), qsrc.end(), 0.0);
+    const double posni[3] = { 0.0 , yi , zi };
+    const double posno[3] = { L , yo , zo };
+    const double rsrc[2] = {0.5*D, 0.5*D};
+    for (int ln = 0; ln <= hierarchy->getFinestLevelNumber(); ++ln)
+    {
+        Pointer<PatchLevel<NDIM> > level = hierarchy->getPatchLevel(ln);
+        for (PatchLevel<NDIM>::Iterator p(level); p; p++)
+        {
+            Pointer<Patch<NDIM> > patch = level->getPatch(p());
+            Pointer<CartesianPatchGeometry<NDIM> > pgeom = patch->getPatchGeometry();
+            if (pgeom->getTouchesRegularBoundary())
+            {
+                Pointer<SideData<NDIM, double> > U_data = patch->getPatchData(U_idx);
+                Pointer<SideData<NDIM, double> > wgt_sc_data = patch->getPatchData(wgt_sc_idx);
+                const Box<NDIM>& patch_box = patch->getBox();
+                const double* const x_lower = pgeom->getXLower();
+                const double* const dx = pgeom->getDx();
+                double dV = 1.0;
+                for (int d = 0; d < NDIM; ++d)
+                {
+                    dV *= dx[d];
+                }
+                double X[NDIM];
+                static const int axis = 0;
+                for (int side = 0; side <= 1; ++side)
+                {
+                    const bool is_lower = side == 0;
+                    if (pgeom->getTouchesRegularBoundary(axis, side))
+                    {
+                       // const double rsrc = d_rsrc[side];
+						//
+                        Vector n;
```

Comment:
```suggestion
                        VectorNd n;
```
This is a good example of something which fails to compile with newer versions of Eigen since we used `app_namespaces.h`.